### ISSUE TITLE
Support Multiple docker files from docker container run

### ISSUE THREAD

**Issue by johngluckmdsol** [hadolint/hadolint#286]
- [] This is a bug report
- [X] This is a feature request
- [X] I searched existing issues before opening this one

### Steps to reproduce the behavior
I have a project I'd like to lint in Travis which contains multiple Dockerfiles.  I'd like to be able to run the hadolint docker container from travis on multiple files without having to hardcode each path.  I have tried mounting a volume but was unable to get this to work.

**Comment by zemanlx** [comment]
Hi you can pass each file in your repo by

```bash
git ls-files --exclude='Dockerfile*' --ignored | xargs --max-lines=1 hadolint
```

This will ensure that you do not get `Dockerfiles` from `.git` or ignored files.

**Comment by johngluckmdsol** [comment]
That doesn't solve my problem.
1. I am running from travis.  Rather than attempt to install it, it would be easier if I can execute hadolint from the docker container.
2. I have several dirs, each of which contains a Dockerfile.  I see what you are saying but that still won't work for me because hadolint only accepts a single file as input.  One solution would be to enable `cat` on the docker image.

**Comment by lorenzo** [comment]
Can you show how you are running it in Travis today?

**Comment by johngluckmdsol** [comment]
What works for us best right now is 
```before_install:
- docker pull hadolint/hadolint

after_script:
- cat */Dockerfile | docker run --rm -i hadolint/hadolint```

We tried many other variants

**Comment by lorenzo** [comment]
Ok, I had a solution like that in mind. How any variant of this would be better for you use case? Is it because you would find it easier to tell travis to run everything from inside the hadolint container instead of having to pull the image manually?

> 1. Furthermore, it's not clear that I can install hadolint in travis without simply checking it out from github

There are many ways you can do this. One is to just download then latest release from GitHub. Here's a one liner:

```bash
curl -s https://api.github.com/repos/hadolint/hadolint/releases/latest\
| grep "hadolint-Linux-x86_64"\
| cut -d : -f 2,3\
| tr -d \"\
| wget -qi -
```

You can also pull the docker container and extract the executable from it:

```
docker run hadolint/hadolint:latest-debian cat /bin/hadolint > hadolint
```



> 2. because hadolint only accepts a single file as input.

Actually hadolint accepts many files as input. Here's an example:

```
docker-files docker run -v `pwd`:/root --rm hadolint/hadolint hadolint /root/envoy/Dockerfile /root/nginx/Dockerfile
/root/envoy/Dockerfile:3 DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
/root/nginx/Dockerfile:3 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
/root/nginx/Dockerfile:3 DL3015 Avoid additional packages by specifying `--no-install-recommends`
```

As you can see I mounted the local directory into `/root` and then manually executed hadolint to pass all the files. The output of hadolint shows in what files the errors happened

**Comment by johngluckmdsol** [comment]
I don't understand this
`docker-files docker run -v `pwd`:/root --rm hadolint/hadolint hadolint /root/envoy/Dockerfile /root/nginx/Dockerfile`.  I'm attempting to avoid specifying paths.  I'm trying to use a wildcard.


This works for me
`docker run hadolint/hadolint:latest-debian cat /bin/hadolint > hadolint`

From there, I simply do
`./hadolint */Dockerfile`

Thanks for the help

**Comment by vinayakkulkarni** [comment]
Is there a way I can run hadolint on `Dockerfile.dev` or `Dockerfile.run` ?
